int16 number
float64[] theta
float64[] dist
float64 min_dist